### DISCUSSION TITLE
Access to pilot output and jdl files of other users

### DISCUSSION THREAD

**Discussion by ernstpijper** [DIRACGrid/DIRAC#6126] [Q&A] [answered]
Hi,

DIRAC py3 server 7.3.17, py3 pilot jobs and py3 diracos2 client

We noticed that users can access the pilot output and jdl input files (and maybe more) of other users. Access to the input sandbox of other users is denied, as it should.

Is this something that should we should allow? Is there maybe an option to turn off this 'feature'?

Thanks,
Ernst

**Comment by ernstpijper** [comment]
We are still wondering about this situation. Did I perhaps not configure something properly?

If in settings if I choose a certain group (not admin), I can also see jobs of users not in that group. I cannot check 
on dirac.egi.eu anymore to see if the same thing is happening there. I used to have access but not anymore. Now i'm just a visitor :(

If this is what it is at the moment, that's ok for me. I just want some clarity about this.

Thanks,
Ernst

**Reply by andresailer** [reply]
In the relevant group, in /Registry/Groups/TheGroup/Properties, is there `JobSharing`?
If you remove that, does it fix the issue? Maybe the JobMonitor (JobManager) services (or more) need to be restarted to reflect the config change.

**Comment by ernstpijper** [comment]
I'm sorry to tell you that I don't use that property.

**Reply by andresailer** [reply]
What properties do you have?

**Comment by ernstpijper** [comment]
I only have Properties = NormalUser

**Comment by andresailer** [comment]
Ok, I understand, you are not talking about the web interface, but the dirac-wms-job-get-jdl commands. Indeed those let you download jdls that do not belong to you. I think this is probably a bug.

**Reply by ernstpijper** [reply]
I'm sorry i realise I wasn't very clear about this. I was indeed talking about the web interface.

**Reply by andresailer** [reply]
You were or were _not_ talking about the web?

**Reply by ernstpijper** [reply]
The problem *is* with the web interface. One user can see the other users pilot job output.

**Reply by andresailer** [reply]
Ok! Can you be more explicit how they get access? So that it can be reproduced? Though they probably go through the same interface as the dirac-wms-job-get-jdl via the web somehow.

**Reply by ernstpijper** [reply]
As a user you go the the pilot monitor in the web interface and, right click on a running or finished job and choose 'Pilot Output'. It then shows you the output of that pilot job.

I used to be able to access dirac.egi.eu. My DN has changed so I can't anymore. Can you perhaps add me again so that I can try to see if I can reproduce it.

**Reply by ernstpijper** [reply]
My new DN is

/DC=org/DC=terena/DC=tcs/C=NL/O=SURF B.V./CN=Ernst Pijper [pijpe001@surf.nl](mailto:pijpe001@surf.nl)

**Reply by andresailer** [reply]
As a user in my instance (iLCDirac) I only see my own jobs (unless JobSharing is part of the group's  properties...) I don't have access to dirac.egi.eu..

Ahh, is it this flag?

https://github.com/DIRACGrid/DIRAC/blob/ed6fe10aacc60a4c35c4ab2d122790ca764a7f08/release.notes#L8285-L8286

https://github.com/DIRACGrid/DIRAC/blob/871cbf92e8365e6376061183c525a57c9196da8d/src/DIRAC/WorkloadManagementSystem/Service/JobMonitoringHandler.py#L330

Can you set this to `False` for your instance?

**Reply by ernstpijper** [reply]
Doesn't seem to help. 

When you go to the Pilot Monitor in your dirac instance as a user, what do you see? As as user I can only see my own jobs in the 'Job Monitor' but I can everybody's pilot jobs in the 'Pilot Monitor'.

**Reply by andresailer** [reply]
Sorry, I missed the "pilot monitor", part, indeed I can see all the pilots and download pilot output

**Reply by fstagni** [reply]
Unless for some reason you are running something quite different, normally ALL pilots jobs are owned by 1 user (and these pilots will match the jobs of all the users). Just for my info, is that user running the pilots yourself?

**Reply by andresailer** [reply]
I can also see the pilots when I am not the owner of all of them in a v7r3 instance, and v8, and also one other person, that doesn't own the pilots can see the pilots in my v7r2 instance.

**Reply by ernstpijper** [reply]
Yes that's what I see. I managed to access dirac.egi.eu again. I can see lots of pilot jobs from different groups, and I can access the corresponding pilot outputs (although sometimes it hangs).

No the user is not running pilot jobs himself.

Ah yes pilots jobs are owned by a single pilot user, although the pilot user can be different for different groups, right.

So what you are saying is the use of pilot jobs means that the pilot job output can be viewed by all users. If that's the case then I suppose that is just the way it is.

**Reply by fstagni** [reply]
I think we have still someone running v7r1 (to see if something changed). @hmiyake would you please check on your instance what you see?

**Reply by hmiyake** [reply]
Today(!) we have moved to v7r2.
Anyway, what we observe is NormalUser can access any JDL (by both web and dirac-wms-job-get-jdl) while sandbox access is limited.
We haven't configured /Services/JobMonitoring/GlobalJobsInfo.

Both dirac-admin-get-job-pilot-output and WebApp allow for NormalUser to retrieve pilot output but the situation is a bit complicated (probably related to CE type and SD option GetPilotOutput (we are enabling this option for various sites))

**Comment by fstagni** [comment, marked as answer]
From @atsareg 

> Right now technically it is not possible to limit user to be able to see only their pilot outputs and JDLs. Pilot outputs can be fully protected from access by "Normal" users without special rights. Other job artefacts like JDLs can be accessed by users in a restricted way but this needs development.

**Reply by ernstpijper** [reply]
Hi Federico,

Thanks for the all effort in answering my questions. I understand now better what the possibilities and limitations are.
